timesketch/lib/utils.py

```diff
@@ -130,14 +148,93 @@ def validate_indices(indices, datastore):
     return [i for i in indices if datastore.client.indices.exists(index=i)]
 
 
-def read_and_validate_csv(file_handle, delimiter=",", mandatory_fields=None):
+def headers_mapping_sanity_check(headers, headers_mapping):
+    """Sanity check for headers mapping
+
+    Args:
+        headers: list of headers found in the CSV file
+        headers_mapping: Python dictionary representing the mapping between
+                         mandatory (key) and existing CSV headers (value).
+
+    Returns: error message if any of the sanity checks fails
+    """
+
+    # 1. The mapping is done only if the mandatory header is missing, and
+    # 2. When a new column is created, a mandatory default value must be set
+    for mand_header in headers_mapping:
+        if mand_header in headers:
+            return "Mapping done only if the mandatory header is missing"
+        if (headers_mapping[mand_header][0] == "New header" and
```

Comment:
here you raise an exception if the condition is true. In the other if-statements I return the error message. Did you mean something like that:
```
if new_header_name == "New header" and not new_header_value:  # equivalent to new_header_value == ""
  error = "..."
  return error
```
?

Reply:
Yes, sorry - I meant raise or return or whatever the behavior of your function is - although as previously stated if you only check for a result, you might want to raise the error in your function direclty instead of return-and-check